attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -72,23 +72,40 @@ pub trait AttestationAPIs {
     /// Bind initdata
     async fn bind_init_data(&self, init_data: &[u8]) -> Result<InitDataResult>;
 
-    fn get_tee_type(&self) -> Tee;
+    fn get_tee_types(&self) -> Vec<Tee>;
 }
 
 /// Attestation agent to provide attestation service.
 pub struct AttestationAgent {
     config: RwLock<Config>,
-    attester: Arc<BoxedAttester>,
+    attesters: Vec<(Tee, Arc<BoxedAttester>)>,
     eventlog: Option<Mutex<EventLog>>,
-    tee: Tee,
+    tees: Vec<Tee>,
 }
 
 impl AttestationAgent {
     pub async fn init(&mut self) -> Result<()> {
         let config = self.config.read().await;
+
         if config.eventlog_config.enable_eventlog {
+            // Identify one attester to use with the EventLog.
+            // Only one attester is expected to support runtime measurements.
+            let runtime_attester = {
+                let mut runtime_attester = None;
+                for (tee, attester) in &self.attesters {
+                    if attester.supports_runtime_measurement() {
+                        info!("Found {:?} for runtime attestation.", tee);
+                        if let Some(_already_found_attester) = runtime_attester {
+                            bail!("Only one attester should support runtime attestation")
+                        }
+                        runtime_attester = Some(attester.clone());
+                    }
+                }
+                runtime_attester.ok_or(anyhow!("No runtime attester found."))?
+            };
```

Comment:
If we reorginaize the code into a hierarchical one, we could avoid the situation judgement here. As `supports_runtime_measurement` actually means `platform/cpu/basic` or something similar